core/src/HPX/Kokkos_HPX.hpp

```diff
@@ -201,6 +202,12 @@ class HPX {
     return impl_get_instance_data().m_instance_id;
   }
 
+  bool is_running() const {
+    // HPX doesn't provide a query function.
+    fence("HPX::is_running");
+    return false;
```

Comment:
@msimberg is this the correct solution?

Reply:
It's not correct, but I guess the question right now is: is it sufficient?

If the primary use case is "fencing" with `while (space.is_running) {}` then this is not going to work (only if `Kokkos_INTERNAL_HPX_ENABLE_ASYNC_DISPATCH=OFF` but that's not the default and shouldn't really be considered for this) for HPX. If a conservative default would be sufficient for now there's an internal counter for the number of kernels submitted to _any_ HPX execution space instance. It would lead to false positives rather than the false negatives that the current `return false` leads to.

The correct solution would require tracking the number of kernels per execution space instance. That shouldn't be hard to add but it's not there at the moment. Is that something you'd like to see added already before this PR is merged or could that be added afterwards?

Finally, note that for the HPX backend `while (space.is_running()) {}` and `space.fence()` would not be 100% equivalent since the latter can yield an HPX thread if the instance is still running a kernel.

Reply:
We discussed that we don't want to promise `while(space.is_running()) {}` to act as a fence but an implementation could deadlock on it (although we test that currently in the unit test). The primary use case is querying if the submitted work has been finished and deciding based on the answer if something else should be done before querying again. 
I would consider `fencing` and returning `false` to be a valid generic implementation that we can improve upon for execution spaces that allow more introspection. I would be happy to see these improvements in another pull request (given reviews agree).

Reply:
Thanks @masterleinad for the explanation. And I'm sorry, I completely missed the `fence` before returning `false`! In that case yes, this works as a fallback, and I can separately add a way to keep track of kernels launched per-instance. I can then update this to remove the fence.